src/Altinn.App.Core/Implementation/PrefillSI.cs

```diff
@@ -99,11 +99,18 @@ IHttpContextAccessor httpContextAccessor
 
             if (profilePrefill != null)
             {
-                var userProfileDict = profilePrefill.ToObject<Dictionary<string, string>>()!;
+                var userProfileDict =
+                    profilePrefill.ToObject<Dictionary<string, string>>()
+                    ?? throw new Exception("Unexpectedly failed to convert profilePrefill JToken to dictionary");
 
                 if (userProfileDict.Count > 0)
                 {
-                    int userId = AuthenticationHelper.GetUserId(_httpContextAccessor.HttpContext!);
+                    var httpContext =
+                        _httpContextAccessor.HttpContext
+                        ?? throw new Exception(
+                            "HttpContext is unexpectedly null, need HttpContext to get current user"
```

Comment:
You wrote this better in a previous location
```suggestion
                            "Could not get HttpContext - must be in a request context to get current user");
```